File: src/main/webapp/src/dto/GameStateDto.ts
import { Round } from "../components/views/game/related/GameStats";
import Phase from "../enum/Phase";
import { CardActionDto } from "./CardActionDto";
import { CardDto } from "./CardDto";
import { GameStatisticsDto } from "./GameStatisticsDto";

type GameStateDto = {
	id: string;
	owner: string;
	currentPlayer: string;
	round: Round;
	hand: CardDto[];
	stats: GameStatisticsDto;
	playedCards: CardActionDto[];
	order: string[];
	phase: Phase;
	validResponseCardIds: number[];
	strongestCardId: number;
};

export type { GameStateDto };

File: src/main/java/brunotot/skullking/web/dto/GameStateDto.java
package brunotot.skullking.web.dto;

import brunotot.skullking.domain.CardAction;
import brunotot.skullking.domain.Phase;
import brunotot.skullking.domain.card.Card;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.Map;

@AllArgsConstructor
@Builder
@Getter
@Setter
public class GameStateDto {
    private String id;
    private String currentPlayer;
    private String owner;
    private List<Card> hand;
    private List<CardAction> playedCards;
    private List<String> order;
    private List<Integer> validResponseCardIds;
    private Map<String, PlayerStatisticsDto> stats;
    private Phase phase;
    private Integer round;
    private Integer strongestCardId;
}
